File: app/src/main/java/com/paradoxo/amadeus/dao/AutorDAO.java
package com.paradoxo.amadeus.dao;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;

import com.paradoxo.amadeus.modelo.Autor;

import java.util.ArrayList;
import java.util.List;

public class AutorDAO {
    
    private static BDGateway bdGateway;

    private static final String TABELA_AUTOR = "Autor";
    
    public AutorDAO(Context context) {
        bdGateway = BDGateway.getInstance(context);
    }

    public void alterar(Autor autor) {
        ContentValues cv = new ContentValues();
        cv.put("nome", autor.getNome());
        bdGateway.getDatabase().update(TABELA_AUTOR, cv, "id=?", new String[]{autor.getId() + ""});
    }

    public List<Autor> listar() {
        List<Autor> autores = new ArrayList<>();
        Cursor cursor = bdGateway.getDatabase().rawQuery("SELECT * FROM autor", null);
        while (cursor.moveToNext()) {
            Autor objAutor = new Autor();
            objAutor.setId(cursor.getInt(cursor.getColumnIndex("id")));
            objAutor.setNome(cursor.getString(cursor.getColumnIndex("nome")));
            autores.add(objAutor);
        }
        cursor.close();
        return autores;
    }

    public Autor buscar(Autor autor) {
        Cursor cursor = bdGateway.getDatabase().rawQuery("SELECT * FROM autor where id =" + autor.getId(), null);
        cursor.moveToNext();

        if (cursor.getCount() > 0) {
            autor.setId(cursor.getInt(cursor.getColumnIndex("id")));
            autor.setNome(cursor.getString(cursor.getColumnIndex("nome")));
        }
        cursor.close();
        return autor;
    }

}

File: app/src/main/java/com/paradoxo/amadeus/dao/AutorDAO.kt
package com.paradoxo.amadeus.dao

import android.content.ContentValues
import android.content.Context
import com.paradoxo.amadeus.modelo.Autor

class AutorDAO(context: Context) {
    private val bdGateway = BDGateway.getInstance(context)

    companion object {
        private const val TABELA_AUTOR = "Autor"
    }

    fun inserirAutor(autor: Autor): Long {
        val cv = ContentValues().apply { put("nome", autor.nome) }
        return bdGateway.database.insert(TABELA_AUTOR, null, cv)
    }

    fun alterar(autor: Autor) {
        val cv = ContentValues().apply { put("nome", autor.nome) }
        bdGateway.database.update(TABELA_AUTOR, cv, "id=?", arrayOf(autor.id.toString()))
    }

    fun listar(): List<Autor> {
        val autores = mutableListOf<Autor>()
        val cursor = bdGateway.database.rawQuery("SELECT * FROM autor", null)
        while (cursor.moveToNext()) {
            autores.add(Autor(
                id = cursor.getInt(cursor.getColumnIndexOrThrow("id")),
                nome = cursor.getString(cursor.getColumnIndexOrThrow("nome"))
            ))
        }
        cursor.close()
        return autores
    }

    fun buscar(autor: Autor): Autor {
        val cursor = bdGateway.database.rawQuery("SELECT * FROM autor where id = ${autor.id}", null)
        cursor.moveToNext()
        if (cursor.count > 0) {
            autor.id = cursor.getInt(cursor.getColumnIndexOrThrow("id"))
            autor.nome = cursor.getString(cursor.getColumnIndexOrThrow("nome"))
        }
        cursor.close()
        return autor
    }

    fun deletarTodosAutores() {
        bdGateway.database.delete(TABELA_AUTOR, null, null)
        bdGateway.database.execSQL("delete from sqlite_sequence where name='autor'")
    }
}
